packages/yarnpkg-shell/sources/index.ts

```diff
@@ -688,11 +690,41 @@ async function executeCommandChain(node: CommandChain, opts: ShellOptions, state
   return await execution.run();
 }
 
+async function executeCommandChain(node: CommandChain, opts: ShellOptions, state: ShellState, {background = false}: {background?: boolean} = {}) {
+  function getColorizer(index: number) {
+    const colors = [`#2E86AB`, `#A23B72`, `#F18F01`, `#C73E1D`, `#CCE2A3`];
+    const colorName = colors[index % colors.length];
+
+    return chalk.hex(colorName);
+  }
+
+  if (background) {
+    const index = state.nextBackgroundJobIndex++;
+    const colorizer = getColorizer(index);
+    const rawPrefix = `[${index}]`;
+    const prefix = colorizer(rawPrefix);
+
+    const {stdout, stderr} = createOutputStreamsWithPrefix(state, {prefix});
+
+    executeCommandChainImpl(node, opts, cloneState(state, {stdout, stderr}))
```

Comment:
How come we never `await` this promise anywhere? Could there be a race condition risk in the tests, where the "background" jobs aren't finished by the time `bufferResult` returns?

Reply:
I intended to store these promises and await them right before `executeShellLine` finishes (i.e. `echo $(sleep 10 & echo foo)` should still sleep for 10 seconds) but I didn't include it in the original prototype and forgot about it before opening this PR, thanks for noticing!